compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1389,8 +1405,14 @@ public void visit(BLangSimpleVariable varNode) {
             varNode.typeNode = rewrite(varNode.typeNode, env);
         }
 
-        // Return if this assignment is not a safe assignment
-        BLangExpression bLangExpression = rewriteExpr(varNode.expr);
+        BLangExpression bLangExpression;
+        if (Symbols.isFlagOn(varNode.symbol.flags, Flags.DEFAULTABLE_PARAM)) {
+            String closureName = CLOSURE_FOR_DEFAULT_VALUE + UNDERSCORE + closureCount++;
```

Comment:
Didn't we discuss to use information from the function (and if required the parameter) to come up with the name for the closure?

This won't work in a backward compatible way.

Reply:
For example, where baz depends on foo and bar, and bar depends on foo.

foo
```ballerina
public function test2(int i = 300, int j = 200) returns int => i + j;
```

bar 
```ballerina
import em/foo;

public function fn() returns int => foo:test2();
```

baz
```ballerina
import em/foo as _;
import em/bar;

public function main() {
    panic error(string `${bar:fn()}`);
}
```

Assume bar used v0.0.1 of foo.

baz uses v0.0.2 of foo which introduced a new function before `test1`, don't your indices change?

v0.0.2 of foo
```ballerina
public function test1(string i = "hello", string j = "world") returns string => i + j;
public function test2(int i = 300, int j = 200) returns int => i + j;
```

Reply:
This crashed for me with

```cmd
[2021-08-30 16:50:37,120] SEVERE {b7a.log.crash} - 'long em.bar.0.main.$closure_for_default_value$_0(io.ballerina.runtime.internal.scheduling.Strand)' 
java.lang.NoSuchMethodError: 'long em.bar.0.main.$closure_for_default_value$_0(io.ballerina.runtime.internal.scheduling.Strand)'
        at em.baz.0.main.fn(main.bal)
        at em.qux.0.main.main(main.bal:5)
        at em.qux.0.$_init.$lambda$main$(qux)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:608)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:333)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:301)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

Reply:
Changed the closure name as  `symbol name + _  + parameter name`